recipes/mimalloc/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+    def _cmake(self):
+        if not self._cmake_i:
+            self._cmake_i = CMake(self)
+            self._cmake_i.definitions["MI_BUILD_TESTS"] = False
+            if self.options.shared:
+                self._cmake_i.definitions["MI_BUILD_SHARED"] = True
+                self._cmake_i.definitions["MI_BUILD_STATIC"] = False
+            else:
+                self._cmake_i.definitions["MI_BUILD_SHARED"] = False
+                self._cmake_i.definitions["MI_BUILD_STATIC"] = True
+            self._cmake_i.configure(source_folder=self._source_subfolder,
+                build_folder=self._build_subfolder)
+        return self._cmake_i
+
+    @property
+    def _msbuild(self):
+        if not self._msbuild_i:
+            self._msbuild_i = MSBuild(self)
+        return self._msbuild_i
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("mimalloc-" + self.version, self._source_subfolder)
+
+    def requirements(self):
+        pass
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        tools.check_min_cppstd(self, "14")
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio":
+            if Version(self.settings.compiler.version) < "15":
+                raise ConanInvalidConfiguration("Visual Studio 15 2017 or newer is required")
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+        if self.settings.compiler == "Visual Studio":
+            target = "mimalloc-override" if self.options.shared else "mimalloc"
+            if Version(self.settings.compiler.version) >= "16":
+                self._msbuild.build(
+                    os.path.join(self._source_subfolder, "ide", "vs2019", "mimalloc.sln"),
+                    targets=[target])
+            else:
+                self._msbuild.build(
+                    os.path.join(self._source_subfolder, "ide", "vs2017", "mimalloc.sln"),
+                    targets=[target])
+        else:
+            self._cmake.build()
+
+    def package(self):
+        if self.settings.compiler == "Visual Studio":
+            self.copy("*.h", dst="include",
+                src=os.path.join(self._source_subfolder, "include"))
+            self.copy("*.lib", dst="lib",
+                src=os.path.join(self._source_subfolder, "out"),
+                keep_path=False)
+            self.copy("*.dll", dst="bin",
+                src=os.path.join(self._source_subfolder, "out"),
+                keep_path=False)
+        else:
+            self._cmake.install()
+            tools.rmdir(os.path.join(self.package_folder, "lib", self._cmake_install_folder, "cmake"))
+
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
- Can you explicitly list the library/libraries, instead of using `tools.collect_libs()`?

Reply:
I thought that `collect_libs()` was the preferred way. I'm listing the libraries from now on.

Reply:
It does not matter for packaging one library, but is important when there are dependencies between the libraries.
Also, it looks nicer to explicitly list the name (but that's subjective).

Reply:
Does it actually work? Because mimalloc installs its libraries in a different prefix.
You might also want to set `self.cpp_info.libs`.